src/index.js

```diff
@@ -199,10 +199,9 @@ function processExternalScripts(options, result) {
  * @param result
  */
 function processMeta(options, result) {
-  result.preprocessed = result.preprocessed.replace(/^---\n(.*\n)*\.\.\. *\n\n?/, meta => {
-    const yaml = meta.replace(/^---\n/, '').replace(/\n\.\.\. *\n?/, '');
+  result.preprocessed = result.preprocessed.replace(/^---\r?\n(.*\r?\n)*\.{3}\s*(\r?\n){2}?/, meta => {
```

Comment:
This changes meaning of the pattern because of the fixed quantifier `{2}`.
`\n\n?` in the original pattern makes the second `\n` optional (with `?`), but `(\r?\n){2}?` _requires two_. I don't think `?` after `{2}` makes any difference because it's fixed.
Changing `{2}?` to `{1,2}` should retain the original intention.

I'd also use non-capturing group `(?:\r?\n)` because it's not used, but that's optional.

Reply:
Thanks!